Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -145,6 +145,16 @@ GC::RootVector<GC::Ref<CSSRule>> Parser::convert_rules(Vector<Rule> const& raw_r
 
             m_declared_namespaces.set(as<CSSNamespaceRule>(*rule).prefix());
             break;
+        case CSSRule::Type::Property: {
+            auto& property_rule = as<CSSPropertyRule>(*rule);
+            if (m_document) {
+                const_cast<DOM::Document*>(m_document.ptr())->set_registered_custom_property(
+                    property_rule.name(),
+                    property_rule
+                );
+            }
+            [[fallthrough]];
+        }
```

Comment:
Unfortunately it's not correct to register the custom property in the Parser. JS can create, remove, or modify `@property` rules, and we want to update our list of registered custom properties whenever those things happen.

It's fine if you just want to leave a FIXME comment here and someone can sort that out later. The only similar case I can think of is `@font-face` and we don't seem to do that correctly so it's not a good example to copy. :sweat_smile: